ast/variables_helper.go

```diff
@@ -3,54 +3,61 @@ package ast
 import "fmt"
 
 func VariableListElementTypesAreHomogenous(variableName string, list []Variable) (Type, error) {
-	listTypes := make(map[Type]struct{})
+	if len(list) == 0 {
+		return TypeInvalid, fmt.Errorf("list %q does not have any elements so cannot determine type.", variableName)
```

Comment:
In the scope of this review I can't claim to know the full impact of this, but this stood out to me as a different behavior. Is this fully understood?